Framework/PythonInterface/plugins/algorithms/HB2AReduce.py

```diff
@@ -0,0 +1,274 @@
+                data = np.append(data, d)
+            except NameError:
+                data = d
+
+        # Get any masked detectors
+        detector_mask = self.get_detector_mask(exp, indir)
+
+        counts = np.array([data['anode{}'.format(n)] for n in range(1,45)])[detector_mask]
+        twotheta = data['2theta']
+        monitor = data['monitor']
+
+        # Get either vcorr file or vanadium data
+        vanadium_count, vanadium_monitor, vcorr = self.get_vanadium(detector_mask,
+                                                                    data['m1'][0], data['colltrans'][0],
+                                                                    exp, indir)
+
+        x = twotheta+self._gaps[:, np.newaxis][detector_mask]
+
+        if self.getProperty("IndividualDetectors").value:
+            # Separate spectrum per anode
+            y, e = self.process(counts, scale, monitor, vanadium_count, vanadium_monitor, vcorr)
+            NSpec=len(x)
+        else:
+            if self.getProperty("BinData").value:
+                # Data binned with bin
+                x, y, e = self.process_binned(counts, x.ravel(), scale, monitor, vanadium_count, vanadium_monitor, vcorr)
+            else:
+                y, e = self.process(counts, scale, monitor, vanadium_count, vanadium_monitor, vcorr)
+            NSpec=1
+
+        createWS_alg = self.createChildAlgorithm("CreateWorkspace")
+        createWS_alg.setProperty("DataX", x)
+        createWS_alg.setProperty("DataY", y)
+        createWS_alg.setProperty("DataE", e)
+        createWS_alg.setProperty("NSpec", NSpec)
+        createWS_alg.setProperty("UnitX", "Degrees")
+        createWS_alg.setProperty("YUnitLabel", "Counts")
+        createWS_alg.setProperty("WorkspaceTitle", str(metadata['scan_title']))
+        createWS_alg.execute()
+        outWS = createWS_alg.getProperty("OutputWorkspace").value
+
+        self.setProperty("OutputWorkspace", outWS)
+
+        self.add_metadata(outWS, metadata, data)
+
+    def get_detector_mask(self, exp, indir):
+        """Returns an anode mask"""
+        detector_mask = np.ones(44, dtype=bool)
+        if len(self.getProperty("ExcludeDetectors").value) == 0:
+            exclude_filename = os.path.join(indir, 'HB2A_{}__exclude_detectors.txt'.format(exp))
+            if os.path.isfile(exclude_filename):
+                with warnings.catch_warnings():
+                    warnings.simplefilter("ignore")
+                    exclude_detectors = np.loadtxt(exclude_filename, ndmin=1, dtype=int)
+            else:
+                exclude_detectors=np.empty(0, dtype=int)
+        else:
+            exclude_detectors = np.array(self.getProperty("ExcludeDetectors").value)
+        if len(exclude_detectors) > 0:
+            logger.notice("Excluding anodes: {}".format(exclude_detectors))
+        detector_mask[exclude_detectors-1] = False
+        return detector_mask
+
+    def get_vanadium(self, detector_mask, m1, colltrans, exp, indir):
```

Comment:
Please document variable names as `m1` and `coltrans` don't really say what they are.